src/keypair/crypto.ts

```diff
@@ -0,0 +1,71 @@
+import secp256k1 = require("bcrypto/lib/secp256k1");
+
+import { IKeypairFull, IKeypairPrivate, IKeypairPublic, KeypairType } from "./types";
+
+const ERR_NOT_IMPLEMENTED = "Not implemented";
+const ERR_INVALID_KEYPAIR_TYPE = "Invalid keypair type";
+
+export function generateKeypair(type: KeypairType): IKeypairFull {
+  let privateKey: Buffer, publicKey: Buffer;
+  switch (type) {
+    case KeypairType.secp256k1:
+      privateKey = secp256k1.privateKeyGenerate();
+      publicKey = secp256k1.publicKeyCreate(privateKey);
+      break;
+    default:
+      throw new Error(ERR_NOT_IMPLEMENTED);
+  }
+  return {
+    type,
+    privateKey,
+    publicKey,
+  };
+}
+
+export function privateKeyVerify(keypair: IKeypairPrivate): boolean {
+  switch (keypair.type) {
+    case KeypairType.secp256k1:
```

Comment:
I would maybe think about some classes to avoid duplicating that switch case. If you wan't to add another keytype you have to add case clause in like 6-7 places

Reply:
:+1:  take another look :)